uaclient/testing/fakes.py

```diff
@@ -34,3 +36,47 @@ def request_url(
         if isinstance(response, Exception):
             raise response
         return response, {"header1": ""}
+
+
+class MockEntitlement(base.UAEntitlement):
+
+    name = "mock-entitlement"
+    title = "Mock Entitlement"
+    description = "Entitlement for testing"
+
+    def __init__(
+        self,
+        cfg=None,
+        assume_yes=False,
+        disable=None,
+        enable=None,
+        applicability_status=None,
+        application_status=None,
+        contract_status=None,
+        allow_beta=False,
+    ):
+        super().__init__(cfg, assume_yes=assume_yes, allow_beta=allow_beta)
+        self._disable = disable
+        self._enable = enable
+        self._applicability_status = applicability_status
+        self._application_status = application_status
+        self._contract_status = contract_status
+
+    def disable(self):
```

Comment:
I don't think we should override by default the `disable` and `_perform_enable` method. I think since those methods perform a more complex logic, the user should explicitly mock then in test, if needed

My concern is that we can hide some testing bits by overriding those methods by default.

Reply:
See above longer answer. The point of MockEntitlement is too be a minimal subclass of UAEntitlement that implements all the abstract methods, including disable and _perform_enable. And MockEntitlement is supposed to not do anything.  I should probably add the purpose of the class in a docstring if we stick with this